src/components/pages/home/elements/WeatherBlock.js

```diff
@@ -1,12 +1,44 @@
 import React, { Component } from 'react';
+import WeatherTitle from './weatherTitle/WeatherTitle';
+import WeathewrAttributes from './weatherAttributes/WeatherAttributes';
+import getMarsWeatherData from '../../../../api/inSightApiClient';
 import '../HomePage.css';
 import PageNavButton from '../../../shared/PageNavButton';
 
 export default class WeatherBlock extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { weather: this.dummyValues() };
+  }
+
+  componentDidMount() {
+    this.getData();
+  }
+
+  getData() {
+    getMarsWeatherData().then(weather => (this.setState({
+      weather: weather.slice(weather.length - 1)[0],
+    })
+    ));
+  }
```

Comment:
The formatting and line spacing of this is a bit confusing, I think I'd go for something like:

```
getData() {
  getMarsWeatherData()
    .then(weather => this.setState({
      weather: weather.slice(weather.length - 1)[0],
    }));
}
```

Just moving the `.then()` onto the next line to make the chain of stuff clearer 😊